development/compiling/compiling_for_android.rst

```diff
@@ -171,80 +178,33 @@ Troubleshooting
 Application not installed
 ~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Android might complain the application is not correctly installed. If
-so, check the following:
+Android might complain the application is not correctly installed.
+If so:
 
 -  Check that the debug keystore is properly generated.
--  Check that jarsigner is from JDK 6, 7 or 8.
+-  Check that the jarsigner executable is from JDK 8.
 
-If it still fails, open a command line and run logcat:
+If it still fails, open a command line and run `logcat <https://developer.android.com/studio/command-line/logcat>`_:
 
 ::
 
-    C:\android-sdk\platform-tools> adb logcat
-
-And check the output while the application is installed. Reason for
-failure should be presented there.
+    adb logcat
 
+Then check the output while the application is installed;
+the error message should be presented there.
 Seek assistance if you can't figure it out.
 
 Application exits immediately
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-If the application runs but exits immediately, there might be one of the
-following reasons:
+If the application runs but exits immediately, this might be due to
+one of the following reasons:
 
 -  Make sure to use export templates that match your editor version; if
    you use a new Godot version, you *have* to update the templates too.
--  libgodot_android.so is not in ``lib/armeabi-v7a`` or ``lib/armeabi``
--  Device does not support armv7 (try compiling yourself for armv6)
--  Device is Intel, and apk is compiled for ARM.
+-  ``libgodot_android.so`` is not in ``lib/armeabi-v7a`` or ``lib/armeabi``.
+-  The device does not support ARMv7; try compiling for ARMv6 by passing
+   ``android_arch=armv6`` to the SCons command line.
+-  The device is x86, but the APK is only compiled for ARM.
```

Comment:
Should also mention `arm64-v8`. But same comment as above, it would be better to rephrase and explain that the exported APK can have support for 1 or more platforms depending on 1) what was compiled in the template 2) what was toggled during export.

Reply:
And as such the "The device does not support ARMv7; try compiling for ARMv6" is also bad advice, as in most cases nowadays, if a device does not support ARMv7 it's likely that it's x86 or ARM64, not obsolete v6. I would simply rephrase it as "The device's architecture does not match the exported one(s). Make sure that your templates were built against that device's architecture, and that the export settings enabled support for that architecture." (feel free to phrase differently)